// liquidify-react/dist/libs/components/chunks/accordion-D6YVHbIZ.mjs
import { Accordion as t } from "@ark-ui/react";
import { m as i, c as s, s as m, a as d, g as p } from "./create-recipe-D2XybSAv.mjs";
import { c as g } from "./create-style-context-8AvEXXie.mjs";
const a = {}, _ = [], I = [
  [
    "root",
    "accordion__root"
  ],
  [
    "item",
    "accordion__item"
  ],
  [
    "trigger",
    "accordion__trigger"
  ],
  [
    "content",
    "accordion__content"
  ]
], l = /* @__PURE__ */ I.map(([o, n]) => [o, d(n, a, p(_, o))]), A = i((o = {}) => Object.fromEntries(l.map(([n, e]) => [n, e.recipeFn(o)]))), r = [], C = (o) => ({ ...a, ...s(o) }), f = /* @__PURE__ */ Object.assign(A, {
  __recipe__: !1,
  __name__: "accordion",
  raw: (o) => o,
  classNameMap: {},
  variantKeys: r,
  variantMap: {},
  splitVariantProps(o) {
    return m(o, r);
  },
  getVariantProps: C
}), { withContext: c } = g(f), V = c(t.Root, "root"), u = c(t.Item, "item"), R = c(t.ItemTrigger, "trigger"), S = c(t.ItemContent, "content"), F = {
  Root: V,
  Item: u,
  ItemTrigger: R,
  ItemContent: S
};
export {
  F as A,
  V as a,
  u as b,
  R as c,
  S as d
};
