// liquidify-react/dist/libs/components/chunks/avatar-dAFYcnOZ.mjs
import { jsxs as V, jsx as e } from "react/jsx-runtime";
import { Avatar as r } from "@ark-ui/react";
import { forwardRef as f } from "react";
import { m as A, a as P, s as u, b as j } from "./create-recipe-D2XybSAv.mjs";
const t = /* @__PURE__ */ P("avatar", {
  size: "md"
}, []), n = {
  size: [
    "sm",
    "md",
    "lg",
    "xl"
  ]
}, o = Object.keys(n), i = /* @__PURE__ */ Object.assign(A(t.recipeFn), {
  __recipe__: !0,
  __name__: "avatar",
  __getCompoundVariantCss__: t.__getCompoundVariantCss__,
  raw: (a) => a,
  variantKeys: o,
  variantMap: n,
  merge(a) {
    return j(this, a);
  },
  splitVariantProps(a) {
    return u(a, o);
  },
  getVariantProps: t.getVariantProps
}), C = f(
  ({ name: a, src: s, fallback: p, size: m, children: c, className: _, ...l }, v) => {
    const [g, d] = i.splitVariantProps({ size: m });
    return /* @__PURE__ */ V(
      r.Root,
      {
        ref: v,
        className: [i(g), _].filter(Boolean).join(" "),
        ...d,
        ...l,
        children: [
          /* @__PURE__ */ e(r.Fallback, { children: p || a?.charAt(0)?.toUpperCase() || "U" }),
          s && /* @__PURE__ */ e(r.Image, { src: s, alt: a || "Avatar" }),
          c
        ]
      }
    );
  }
);
C.displayName = "Avatar";
export {
  C as A
};
