// liquidify-react/dist/libs/components/chunks/carousel-Cy2nfZRz.mjs
import { Carousel as o } from "@ark-ui/react";
import { m as c, c as n, s as u, a as l, g as p } from "./create-recipe-D2XybSAv.mjs";
import { c as g } from "./create-style-context-8AvEXXie.mjs";
const s = {}, m = [], _ = [
  [
    "root",
    "carousel__root"
  ],
  [
    "viewport",
    "carousel__viewport"
  ],
  [
    "itemGroup",
    "carousel__itemGroup"
  ],
  [
    "item",
    "carousel__item"
  ],
  [
    "nextTrigger",
    "carousel__nextTrigger"
  ],
  [
    "prevTrigger",
    "carousel__prevTrigger"
  ],
  [
    "indicatorGroup",
    "carousel__indicatorGroup"
  ],
  [
    "indicator",
    "carousel__indicator"
  ]
], C = /* @__PURE__ */ _.map(([r, t]) => [r, l(t, s, p(m, r))]), d = c((r = {}) => Object.fromEntries(C.map(([t, i]) => [t, i.recipeFn(r)]))), a = [], v = (r) => ({ ...s, ...n(r) }), G = /* @__PURE__ */ Object.assign(d, {
  __recipe__: !1,
  __name__: "carousel",
  raw: (r) => r,
  classNameMap: {},
  variantKeys: a,
  variantMap: {},
  splitVariantProps(r) {
    return u(r, a);
  },
  getVariantProps: v
}), { withRootProvider: I, withContext: e } = g(G), T = I(o.Root), x = e(o.Control, "viewport"), f = e(o.ItemGroup, "itemGroup"), P = e(o.Item, "item"), w = e(o.NextTrigger, "nextTrigger"), V = e(o.PrevTrigger, "prevTrigger"), R = e(o.IndicatorGroup, "indicatorGroup"), N = e(o.Indicator, "indicator"), y = {
  Root: T,
  Control: x,
  ItemGroup: f,
  Item: P,
  NextTrigger: w,
  PrevTrigger: V,
  IndicatorGroup: R,
  Indicator: N
};
export {
  y as C,
  T as a,
  x as b,
  f as c,
  P as d,
  w as e,
  V as f,
  R as g,
  N as h
};
