// liquidify-react/dist/libs/components/chunks/combobox-DG6DwCnG.mjs
import { Combobox as t } from "@ark-ui/react";
import { m, c as i, s as a, a as c, g as x } from "./create-recipe-D2XybSAv.mjs";
import { c as p } from "./create-style-context-8AvEXXie.mjs";
const s = {}, g = [], C = [
  [
    "root",
    "combobox__root"
  ],
  [
    "trigger",
    "combobox__trigger"
  ],
  [
    "input",
    "combobox__input"
  ],
  [
    "positioner",
    "combobox__positioner"
  ],
  [
    "content",
    "combobox__content"
  ],
  [
    "item",
    "combobox__item"
  ],
  [
    "itemText",
    "combobox__itemText"
  ]
], _ = /* @__PURE__ */ C.map(([o, n]) => [o, c(n, s, x(g, o))]), T = m((o = {}) => Object.fromEntries(_.map(([n, b]) => [n, b.recipeFn(o)]))), r = [], l = (o) => ({ ...s, ...i(o) }), u = /* @__PURE__ */ Object.assign(T, {
  __recipe__: !1,
  __name__: "combobox",
  raw: (o) => o,
  classNameMap: {},
  variantKeys: r,
  variantMap: {},
  splitVariantProps(o) {
    return a(o, r);
  },
  getVariantProps: l
}), { withRootProvider: I, withContext: e } = p(u), f = I(t.Root), P = e(t.Trigger, "trigger"), V = e(t.Input, "input"), R = e(t.Positioner, "positioner"), d = e(t.Content, "content"), S = e(t.Item, "item"), h = e(t.ItemText, "itemText"), v = e(t.ClearTrigger, "trigger"), j = {
  Root: f,
  Trigger: P,
  Input: V,
  Positioner: R,
  Content: d,
  Item: S,
  ItemText: h,
  ClearTrigger: v
};
export {
  j as C,
  f as a,
  P as b,
  V as c,
  R as d,
  d as e,
  S as f,
  h as g,
  v as h
};
