// liquidify-react/dist/libs/components/chunks/create-recipe-D2XybSAv.mjs
function f(r) {
  return typeof r == "object" && r != null && !Array.isArray(r);
}
var D = (r) => typeof r == "object" && r !== null;
function C(r) {
  return Object.fromEntries(Object.entries(r ?? {}).filter(([t, o]) => o !== void 0));
}
var V = (r) => r === "base";
function N(r) {
  return r.slice().filter((t) => !V(t));
}
function S(r) {
  return String.fromCharCode(r + (r > 25 ? 39 : 97));
}
function $(r) {
  let t = "", o;
  for (o = Math.abs(r); o > 52; o = o / 52 | 0) t = S(o % 52) + t;
  return S(o % 52) + t;
}
function G(r, t) {
  let o = t.length;
  for (; o; ) r = r * 33 ^ t.charCodeAt(--o);
  return r;
}
function H(r) {
  return $(G(5381, r) >>> 0);
}
var w = /\s*!(important)?/i;
function q(r) {
  return typeof r == "string" ? w.test(r) : !1;
}
function K(r) {
  return typeof r == "string" ? r.replace(w, "").trim() : r;
}
function m(r) {
  return typeof r == "string" ? r.replaceAll(" ", "_") : r;
}
var h = (r) => {
  const t = /* @__PURE__ */ new Map();
  return (...n) => {
    const i = JSON.stringify(n);
    if (t.has(i))
      return t.get(i);
    const e = r(...n);
    return t.set(i, e), e;
  };
}, Z = /* @__PURE__ */ new Set(["__proto__", "constructor", "prototype"]);
function R(...r) {
  return r.reduce((t, o) => (o && Object.keys(o).forEach((n) => {
    if (Z.has(n)) return;
    const i = t[n], e = o[n];
    f(i) && f(e) ? t[n] = R(i, e) : t[n] = e;
  }), t), {});
}
var U = (r) => r != null;
function B(r, t, o = {}) {
  const { stop: n, getKey: i } = o;
  function e(s, a = []) {
    if (D(s)) {
      const d = {};
      for (const [l, b] of Object.entries(s)) {
        const c = i?.(l, b) ?? l, g = [...a, c];
        if (n?.(s, g))
          return t(s, a);
        const p = e(b, g);
        U(p) && (d[c] = p);
      }
      return d;
    }
    return t(s, a);
  }
  return e(r);
}
function J(r, t) {
  return r.reduce(
    (o, n, i) => {
      const e = t[i];
      return n != null && (o[e] = n), o;
    },
    {}
  );
}
function E(r, t, o = !0) {
  const { utility: n, conditions: i } = t, { hasShorthand: e, resolveShorthand: s } = n;
  return B(
    r,
    (a) => Array.isArray(a) ? J(a, i.breakpoints.keys) : a,
    {
      stop: (a) => Array.isArray(a),
      getKey: o ? (a) => e ? s(a) : a : void 0
    }
  );
}
var Q = {
  shift: (r) => r,
  finalize: (r) => r,
  breakpoints: { keys: [] }
}, rr = (r) => typeof r == "string" ? r.replaceAll(/[\n\s]+/g, " ") : r;
function P(r) {
  const { utility: t, hash: o, conditions: n = Q } = r, i = (s) => [t.prefix, s].filter(Boolean).join("-"), e = (s, a) => {
    let d;
    if (o) {
      const l = [...n.finalize(s), a];
      d = i(t.toHash(l, H));
    } else
      d = [...n.finalize(s), i(a)].join(":");
    return d;
  };
  return h(({ base: s, ...a } = {}) => {
    const d = Object.assign(a, s), l = E(d, r), b = /* @__PURE__ */ new Set();
    return B(l, (c, g) => {
      if (c == null) return;
      const p = q(c), [L, ...F] = n.shift(g), X = N(F), Y = t.transform(L, K(rr(c)));
      let u = e(X, Y.className);
      p && (u = `${u}!`), b.add(u);
    }), Array.from(b).join(" ");
  });
}
function tr(...r) {
  return r.flat().filter((t) => f(t) && Object.keys(C(t)).length > 0);
}
function or(r) {
  function t(i) {
    const e = tr(...i);
    return e.length === 1 ? e : e.map((s) => E(s, r));
  }
  function o(...i) {
    return R(...t(i));
  }
  function n(...i) {
    return Object.assign({}, ...t(i));
  }
  return { mergeCss: h(o), assignCss: n };
}
var nr = /([A-Z])/g, er = /^ms-/, ir = h((r) => r.startsWith("--") ? r : r.replace(nr, "-$1").replace(er, "-ms-").toLowerCase()), sr = "cm,mm,Q,in,pc,pt,px,em,ex,ch,rem,lh,rlh,vw,vh,vmin,vmax,vb,vi,svw,svh,lvw,lvh,dvw,dvh,cqw,cqh,cqi,cqb,cqmin,cqmax,%";
`${sr.split(",").join("|")}`;
var fr = (r, t) => r.filter((o) => o.css[t]).map((o) => ({ ...o, css: o.css[t] }));
function ar(r, ...t) {
  const o = Object.getOwnPropertyDescriptors(r), n = Object.keys(o), i = (s) => {
    const a = {};
    for (let d = 0; d < s.length; d++) {
      const l = s[d];
      o[l] && (Object.defineProperty(a, l, o[l]), delete o[l]);
    }
    return a;
  }, e = (s) => i(Array.isArray(s) ? s : n.filter(s));
  return t.map(e).concat(i(n));
}
var y = (...r) => {
  const t = r.reduce((o, n) => (n && n.forEach((i) => o.add(i)), o), /* @__PURE__ */ new Set([]));
  return Array.from(t);
};
const dr = "_hover,_focus,_focusWithin,_focusVisible,_disabled,_active,_visited,_target,_readOnly,_readWrite,_empty,_checked,_enabled,_expanded,_highlighted,_complete,_incomplete,_dragging,_before,_after,_firstLetter,_firstLine,_marker,_selection,_file,_backdrop,_first,_last,_only,_even,_odd,_firstOfType,_lastOfType,_onlyOfType,_peerFocus,_peerHover,_peerActive,_peerFocusWithin,_peerFocusVisible,_peerDisabled,_peerChecked,_peerInvalid,_peerExpanded,_peerPlaceholderShown,_groupFocus,_groupHover,_groupActive,_groupFocusWithin,_groupFocusVisible,_groupDisabled,_groupChecked,_groupExpanded,_groupInvalid,_indeterminate,_required,_valid,_invalid,_autofill,_inRange,_outOfRange,_placeholder,_placeholderShown,_pressed,_selected,_grabbed,_underValue,_overValue,_atValue,_default,_optional,_open,_closed,_fullscreen,_loading,_hidden,_current,_currentPage,_currentStep,_today,_unavailable,_rangeStart,_rangeEnd,_now,_topmost,_motionReduce,_motionSafe,_print,_landscape,_portrait,_dark,_light,_osDark,_osLight,_highContrast,_lessContrast,_moreContrast,_ltr,_rtl,_scrollbar,_scrollbarThumb,_scrollbarTrack,_horizontal,_vertical,_icon,_starting,_noscript,_invertedColors,base", I = new Set(dr.split(",")), lr = /^@|&|&$/;
function k(r) {
  return I.has(r) || lr.test(r);
}
const cr = /^_/, br = /&|@/;
function O(r) {
  return r.map((t) => I.has(t) ? t.replace(cr, "") : br.test(t) ? `[${m(t.trim())}]` : t);
}
function T(r) {
  return r.sort((t, o) => {
    const n = k(t), i = k(o);
    return n && !i ? 1 : !n && i ? -1 : 0;
  });
}
const gr = "aspectRatio:asp,boxDecorationBreak:bx-db,zIndex:z,boxSizing:bx-s,objectPosition:obj-p,objectFit:obj-f,overscrollBehavior:ovs-b,overscrollBehaviorX:ovs-bx,overscrollBehaviorY:ovs-by,position:pos/1,top:top,left:left,inset:inset,insetInline:inset-x/insetX,insetBlock:inset-y/insetY,insetBlockEnd:inset-be,insetBlockStart:inset-bs,insetInlineEnd:inset-e/insetEnd/end,insetInlineStart:inset-s/insetStart/start,right:right,bottom:bottom,float:float,visibility:vis,display:d,hideFrom:hide,hideBelow:show,flexBasis:flex-b,flex:flex,flexDirection:flex-d/flexDir,flexGrow:flex-g,flexShrink:flex-sh,gridTemplateColumns:grid-tc,gridTemplateRows:grid-tr,gridColumn:grid-c,gridRow:grid-r,gridColumnStart:grid-cs,gridColumnEnd:grid-ce,gridAutoFlow:grid-af,gridAutoColumns:grid-ac,gridAutoRows:grid-ar,gap:gap,gridGap:grid-g,gridRowGap:grid-rg,gridColumnGap:grid-cg,rowGap:rg,columnGap:cg,justifyContent:jc,alignContent:ac,alignItems:ai,alignSelf:as,padding:p/1,paddingLeft:pl/1,paddingRight:pr/1,paddingTop:pt/1,paddingBottom:pb/1,paddingBlock:py/1/paddingY,paddingBlockEnd:pbe,paddingBlockStart:pbs,paddingInline:px/paddingX/1,paddingInlineEnd:pe/1/paddingEnd,paddingInlineStart:ps/1/paddingStart,marginLeft:ml/1,marginRight:mr/1,marginTop:mt/1,marginBottom:mb/1,margin:m/1,marginBlock:my/1/marginY,marginBlockEnd:mbe,marginBlockStart:mbs,marginInline:mx/1/marginX,marginInlineEnd:me/1/marginEnd,marginInlineStart:ms/1/marginStart,spaceX:sx,spaceY:sy,outlineWidth:ring-w/ringWidth,outlineColor:ring-c/ringColor,outline:ring/1,outlineOffset:ring-o/ringOffset,focusRing:focus-ring,focusVisibleRing:focus-v-ring,focusRingColor:focus-ring-c,focusRingOffset:focus-ring-o,focusRingWidth:focus-ring-w,focusRingStyle:focus-ring-s,divideX:dvd-x,divideY:dvd-y,divideColor:dvd-c,divideStyle:dvd-s,width:w/1,inlineSize:w-is,minWidth:min-w/minW,minInlineSize:min-w-is,maxWidth:max-w/maxW,maxInlineSize:max-w-is,height:h/1,blockSize:h-bs,minHeight:min-h/minH,minBlockSize:min-h-bs,maxHeight:max-h/maxH,maxBlockSize:max-b,boxSize:size,color:c,fontFamily:ff,fontSize:fs,fontSizeAdjust:fs-a,fontPalette:fp,fontKerning:fk,fontFeatureSettings:ff-s,fontWeight:fw,fontSmoothing:fsmt,fontVariant:fv,fontVariantAlternates:fv-alt,fontVariantCaps:fv-caps,fontVariationSettings:fv-s,fontVariantNumeric:fv-num,letterSpacing:ls,lineHeight:lh,textAlign:ta,textDecoration:td,textDecorationColor:td-c,textEmphasisColor:te-c,textDecorationStyle:td-s,textDecorationThickness:td-t,textUnderlineOffset:tu-o,textTransform:tt,textIndent:ti,textShadow:tsh,textShadowColor:tsh-c/textShadowColor,textOverflow:tov,verticalAlign:va,wordBreak:wb,textWrap:tw,truncate:trunc,lineClamp:lc,listStyleType:li-t,listStylePosition:li-pos,listStyleImage:li-img,listStyle:li-s,backgroundPosition:bg-p/bgPosition,backgroundPositionX:bg-p-x/bgPositionX,backgroundPositionY:bg-p-y/bgPositionY,backgroundAttachment:bg-a/bgAttachment,backgroundClip:bg-cp/bgClip,background:bg/1,backgroundColor:bg-c/bgColor,backgroundOrigin:bg-o/bgOrigin,backgroundImage:bg-i/bgImage,backgroundRepeat:bg-r/bgRepeat,backgroundBlendMode:bg-bm/bgBlendMode,backgroundSize:bg-s/bgSize,backgroundGradient:bg-grad/bgGradient,backgroundLinear:bg-linear/bgLinear,backgroundRadial:bg-radial/bgRadial,backgroundConic:bg-conic/bgConic,textGradient:txt-grad,gradientFromPosition:grad-from-pos,gradientToPosition:grad-to-pos,gradientFrom:grad-from,gradientTo:grad-to,gradientVia:grad-via,gradientViaPosition:grad-via-pos,borderRadius:bdr/rounded,borderTopLeftRadius:bdr-tl/roundedTopLeft,borderTopRightRadius:bdr-tr/roundedTopRight,borderBottomRightRadius:bdr-br/roundedBottomRight,borderBottomLeftRadius:bdr-bl/roundedBottomLeft,borderTopRadius:bdr-t/roundedTop,borderRightRadius:bdr-r/roundedRight,borderBottomRadius:bdr-b/roundedBottom,borderLeftRadius:bdr-l/roundedLeft,borderStartStartRadius:bdr-ss/roundedStartStart,borderStartEndRadius:bdr-se/roundedStartEnd,borderStartRadius:bdr-s/roundedStart,borderEndStartRadius:bdr-es/roundedEndStart,borderEndEndRadius:bdr-ee/roundedEndEnd,borderEndRadius:bdr-e/roundedEnd,border:bd,borderWidth:bd-w,borderTopWidth:bd-t-w,borderLeftWidth:bd-l-w,borderRightWidth:bd-r-w,borderBottomWidth:bd-b-w,borderBlockStartWidth:bd-bs-w,borderBlockEndWidth:bd-be-w,borderColor:bd-c,borderInline:bd-x/borderX,borderInlineWidth:bd-x-w/borderXWidth,borderInlineColor:bd-x-c/borderXColor,borderBlock:bd-y/borderY,borderBlockWidth:bd-y-w/borderYWidth,borderBlockColor:bd-y-c/borderYColor,borderLeft:bd-l,borderLeftColor:bd-l-c,borderInlineStart:bd-s/borderStart,borderInlineStartWidth:bd-s-w/borderStartWidth,borderInlineStartColor:bd-s-c/borderStartColor,borderRight:bd-r,borderRightColor:bd-r-c,borderInlineEnd:bd-e/borderEnd,borderInlineEndWidth:bd-e-w/borderEndWidth,borderInlineEndColor:bd-e-c/borderEndColor,borderTop:bd-t,borderTopColor:bd-t-c,borderBottom:bd-b,borderBottomColor:bd-b-c,borderBlockEnd:bd-be,borderBlockEndColor:bd-be-c,borderBlockStart:bd-bs,borderBlockStartColor:bd-bs-c,opacity:op,boxShadow:bx-sh/shadow,boxShadowColor:bx-sh-c/shadowColor,mixBlendMode:mix-bm,filter:filter,brightness:brightness,contrast:contrast,grayscale:grayscale,hueRotate:hue-rotate,invert:invert,saturate:saturate,sepia:sepia,dropShadow:drop-shadow,blur:blur,backdropFilter:bkdp,backdropBlur:bkdp-blur,backdropBrightness:bkdp-brightness,backdropContrast:bkdp-contrast,backdropGrayscale:bkdp-grayscale,backdropHueRotate:bkdp-hue-rotate,backdropInvert:bkdp-invert,backdropOpacity:bkdp-opacity,backdropSaturate:bkdp-saturate,backdropSepia:bkdp-sepia,borderCollapse:bd-cl,borderSpacing:bd-sp,borderSpacingX:bd-sx,borderSpacingY:bd-sy,tableLayout:tbl,transitionTimingFunction:trs-tmf,transitionDelay:trs-dly,transitionDuration:trs-dur,transitionProperty:trs-prop,transition:trs,animation:anim,animationName:anim-n,animationTimingFunction:anim-tmf,animationDuration:anim-dur,animationDelay:anim-dly,animationPlayState:anim-ps,animationComposition:anim-comp,animationFillMode:anim-fm,animationDirection:anim-dir,animationIterationCount:anim-ic,animationRange:anim-r,animationState:anim-s,animationRangeStart:anim-rs,animationRangeEnd:anim-re,animationTimeline:anim-tl,transformOrigin:trf-o,transformBox:trf-b,transformStyle:trf-s,transform:trf,rotate:rotate,rotateX:rotate-x,rotateY:rotate-y,rotateZ:rotate-z,scale:scale,scaleX:scale-x,scaleY:scale-y,translate:translate,translateX:translate-x/x,translateY:translate-y/y,translateZ:translate-z/z,accentColor:ac-c,caretColor:ca-c,scrollBehavior:scr-bhv,scrollbar:scr-bar,scrollbarColor:scr-bar-c,scrollbarGutter:scr-bar-g,scrollbarWidth:scr-bar-w,scrollMargin:scr-m,scrollMarginLeft:scr-ml,scrollMarginRight:scr-mr,scrollMarginTop:scr-mt,scrollMarginBottom:scr-mb,scrollMarginBlock:scr-my/scrollMarginY,scrollMarginBlockEnd:scr-mbe,scrollMarginBlockStart:scr-mbt,scrollMarginInline:scr-mx/scrollMarginX,scrollMarginInlineEnd:scr-me,scrollMarginInlineStart:scr-ms,scrollPadding:scr-p,scrollPaddingBlock:scr-py/scrollPaddingY,scrollPaddingBlockStart:scr-pbs,scrollPaddingBlockEnd:scr-pbe,scrollPaddingInline:scr-px/scrollPaddingX,scrollPaddingInlineEnd:scr-pe,scrollPaddingInlineStart:scr-ps,scrollPaddingLeft:scr-pl,scrollPaddingRight:scr-pr,scrollPaddingTop:scr-pt,scrollPaddingBottom:scr-pb,scrollSnapAlign:scr-sa,scrollSnapStop:scrs-s,scrollSnapType:scrs-t,scrollSnapStrictness:scrs-strt,scrollSnapMargin:scrs-m,scrollSnapMarginTop:scrs-mt,scrollSnapMarginBottom:scrs-mb,scrollSnapMarginLeft:scrs-ml,scrollSnapMarginRight:scrs-mr,scrollSnapCoordinate:scrs-c,scrollSnapDestination:scrs-d,scrollSnapPointsX:scrs-px,scrollSnapPointsY:scrs-py,scrollSnapTypeX:scrs-tx,scrollSnapTypeY:scrs-ty,scrollTimeline:scrtl,scrollTimelineAxis:scrtl-a,scrollTimelineName:scrtl-n,touchAction:tch-a,userSelect:us,overflow:ov,overflowWrap:ov-wrap,overflowX:ov-x,overflowY:ov-y,overflowAnchor:ov-a,overflowBlock:ov-b,overflowInline:ov-i,overflowClipBox:ovcp-bx,overflowClipMargin:ovcp-m,overscrollBehaviorBlock:ovs-bb,overscrollBehaviorInline:ovs-bi,fill:fill,stroke:stk,strokeWidth:stk-w,strokeDasharray:stk-dsh,strokeDashoffset:stk-do,strokeLinecap:stk-lc,strokeLinejoin:stk-lj,strokeMiterlimit:stk-ml,strokeOpacity:stk-op,srOnly:sr,debug:debug,appearance:ap,backfaceVisibility:bfv,clipPath:cp-path,hyphens:hy,mask:msk,maskImage:msk-i,maskSize:msk-s,textSizeAdjust:txt-adj,container:cq,containerName:cq-n,containerType:cq-t,cursor:cursor", j = /* @__PURE__ */ new Map(), M = /* @__PURE__ */ new Map();
gr.split(",").forEach((r) => {
  const [t, o] = r.split(":"), [n, ...i] = o.split("/");
  j.set(t, n), i.length && i.forEach((e) => {
    M.set(e === "1" ? n : e, t);
  });
});
const v = (r) => M.get(r) || r, W = {
  conditions: {
    shift: T,
    finalize: O,
    breakpoints: { keys: ["base"] }
  },
  utility: {
    transform: (r, t) => {
      const o = v(r);
      return { className: `${j.get(o) || ir(o)}_${m(t)}` };
    },
    hasShorthand: !0,
    toHash: (r, t) => t(r.join(":")),
    resolveShorthand: v
  }
}, pr = P(W), z = (...r) => pr(_(...r));
z.raw = (...r) => _(...r);
const { mergeCss: _ } = or(W);
function x(r, t) {
  let o = {};
  return r.forEach((n) => {
    Object.entries(n).every(([e, s]) => e === "css" ? !0 : (Array.isArray(s) ? s : [s]).some((d) => t[e] === d)) && (o = _(o, n.css));
  }), o;
}
function ur(r, t, o, n) {
  if (t.length > 0 && typeof o?.[n] == "object")
    throw new Error(`[recipe:${r}:${n}] Conditions are not supported when using compound variants.`);
}
function A() {
  let r = "", t = 0, o;
  for (; t < arguments.length; )
    (o = arguments[t++]) && typeof o == "string" && (r && (r += " "), r += o);
  return r;
}
const mr = (r, t, o) => {
  const n = (e) => ({
    [r]: "__ignore__",
    ...t,
    ...C(e)
  });
  return {
    recipeFn: (e, s = !0) => {
      const d = P({
        conditions: {
          shift: T,
          finalize: O,
          breakpoints: { keys: ["base"] }
        },
        utility: {
          toHash: (b, c) => c(b.join(":")),
          transform: (b, c) => (ur(r, o, e, b), c === "__ignore__" ? { className: r } : (c = m(c), { className: `${r}--${b}_${c}` }))
        }
      }), l = n(e);
      if (s) {
        const b = x(o, l);
        return A(d(l), z(b));
      }
      return d(l);
    },
    getVariantProps: n,
    __getCompoundVariantCss__: (e) => x(o, n(e))
  };
}, hr = (r, t) => {
  if (r && !t) return r;
  if (!r && t) return t;
  const o = (...e) => A(r(...e), t(...e)), n = y(r.variantKeys, t.variantKeys), i = n.reduce((e, s) => (e[s] = y(r.variantMap[s], t.variantMap[s]), e), {});
  return Object.assign(o, {
    __recipe__: !0,
    __name__: `${r.__name__} ${t.__name__}`,
    raw: (e) => e,
    variantKeys: n,
    variantMap: i,
    splitVariantProps(e) {
      return ar(e, n);
    }
  });
};
export {
  mr as a,
  hr as b,
  C as c,
  fr as g,
  h as m,
  ar as s
};
