// liquidify-react/dist/libs/components/chunks/datePicker-DY2zQeBN.mjs
import { DatePicker as e } from "@ark-ui/react";
import { m as n, c as s, s as l, a as p, g as d } from "./create-recipe-D2XybSAv.mjs";
import { c as b } from "./create-style-context-8AvEXXie.mjs";
const i = {}, k = [], P = [
  ["root", "date-picker__root"],
  ["trigger", "date-picker__trigger"],
  ["input", "date-picker__input"],
  ["positioner", "date-picker__positioner"],
  ["content", "date-picker__content"],
  ["table", "date-picker__table"],
  ["tableHead", "date-picker__tableHead"],
  ["tableBody", "date-picker__tableBody"],
  ["tableRow", "date-picker__tableRow"],
  ["tableCell", "date-picker__tableCell"]
], _ = /* @__PURE__ */ P.map(([t, r]) => [
  t,
  p(
    r,
    i,
    d(k, t)
  )
]), g = n((t = {}) => Object.fromEntries(
  _.map(([r, c]) => [r, c.recipeFn(t)])
)), o = [], T = (t) => ({ ...i, ...s(t) }), m = /* @__PURE__ */ Object.assign(g, {
  __recipe__: !1,
  __name__: "datePicker",
  raw: (t) => t,
  classNameMap: {},
  variantKeys: o,
  variantMap: {},
  splitVariantProps(t) {
    return l(t, o);
  },
  getVariantProps: T
}), { withRootProvider: C, withContext: a } = b(m), D = C(e.Root), u = a(e.Trigger, "trigger"), R = a(e.Input, "input"), w = a(e.Positioner, "positioner"), y = a(e.Content, "content"), f = a(e.Table, "table"), B = a(e.TableHead, "tableHead"), H = a(e.TableBody, "tableBody"), V = a(e.TableRow, "tableRow"), S = a(e.TableCell, "tableCell"), h = a(e.ClearTrigger, "trigger"), F = {
  Root: D,
  Trigger: u,
  Input: R,
  Positioner: w,
  Content: y,
  Table: f,
  TableHead: B,
  TableBody: H,
  TableRow: V,
  TableCell: S,
  ClearTrigger: h
};
export {
  F as D,
  D as a,
  u as b,
  R as c,
  w as d,
  y as e,
  f,
  B as g,
  H as h,
  V as i,
  S as j,
  h as k
};
