// liquidify-react/dist/libs/components/chunks/dialog-CPfGeOf7.mjs
import { Dialog as t } from "@ark-ui/react";
import { m as n, c as g, s as c, a as l, g as p } from "./create-recipe-D2XybSAv.mjs";
import { c as d } from "./create-style-context-8AvEXXie.mjs";
const e = {}, _ = [], m = [
  [
    "backdrop",
    "dialog__backdrop"
  ],
  [
    "positioner",
    "dialog__positioner"
  ],
  [
    "content",
    "dialog__content"
  ],
  [
    "title",
    "dialog__title"
  ],
  [
    "description",
    "dialog__description"
  ],
  [
    "trigger",
    "dialog__trigger"
  ],
  [
    "closeTrigger",
    "dialog__closeTrigger"
  ]
], D = /* @__PURE__ */ m.map(([o, a]) => [o, l(a, e, p(_, o))]), T = n((o = {}) => Object.fromEntries(D.map(([a, s]) => [a, s.recipeFn(o)]))), r = [], C = (o) => ({ ...e, ...g(o) }), f = /* @__PURE__ */ Object.assign(T, {
  __recipe__: !1,
  __name__: "dialog",
  raw: (o) => o,
  classNameMap: {},
  variantKeys: r,
  variantMap: {},
  splitVariantProps(o) {
    return c(o, r);
  },
  getVariantProps: C
}), { withRootProvider: P, withContext: i } = d(f), b = P(t.Root), k = i(t.Backdrop, "backdrop"), V = i(t.Positioner, "positioner"), u = i(t.Content, "content"), R = i(t.Title, "title"), S = i(t.Description, "description"), h = i(t.Trigger, "trigger"), v = i(t.CloseTrigger, "closeTrigger"), B = {
  Root: b,
  Backdrop: k,
  Positioner: V,
  Content: u,
  Title: R,
  Description: S,
  Trigger: h,
  CloseTrigger: v
};
export {
  B as D,
  b as a,
  k as b,
  V as c,
  u as d,
  R as e,
  S as f,
  h as g,
  v as h
};
