// liquidify-react/dist/libs/components/chunks/menu-ClOLbxd5.mjs
import { Menu as e } from "@ark-ui/react";
import { m as i, c as m, s as c, a as u, g as p } from "./create-recipe-D2XybSAv.mjs";
import { c as _ } from "./create-style-context-8AvEXXie.mjs";
const a = {}, g = [], M = [
  [
    "trigger",
    "menu__trigger"
  ],
  [
    "positioner",
    "menu__positioner"
  ],
  [
    "content",
    "menu__content"
  ],
  [
    "item",
    "menu__item"
  ],
  [
    "itemText",
    "menu__itemText"
  ],
  [
    "separator",
    "menu__separator"
  ]
], x = /* @__PURE__ */ M.map(([t, o]) => [t, u(o, a, p(g, t))]), T = i((t = {}) => Object.fromEntries(x.map(([o, s]) => [o, s.recipeFn(t)]))), r = [], l = (t) => ({ ...a, ...m(t) }), f = /* @__PURE__ */ Object.assign(T, {
  __recipe__: !1,
  __name__: "menu",
  raw: (t) => t,
  classNameMap: {},
  variantKeys: r,
  variantMap: {},
  splitVariantProps(t) {
    return c(t, r);
  },
  getVariantProps: l
}), { withRootProvider: C, withContext: n } = _(f), P = C(e.Root), S = n(e.Trigger, "trigger"), I = n(e.Positioner, "positioner"), V = n(e.Content, "content"), R = n(e.Item, "item"), d = n(e.ItemText, "itemText"), b = n(e.Separator, "separator"), F = {
  Root: P,
  Trigger: S,
  Positioner: I,
  Content: V,
  Item: R,
  ItemText: d,
  Separator: b
};
export {
  F as M,
  P as a,
  S as b,
  I as c,
  V as d,
  R as e,
  d as f,
  b as g
};
