// liquidify-react/dist/libs/components/chunks/numberInput-CpMyWkYf.mjs
import { NumberInput as r } from "@ark-ui/react";
import { m as i, c as a, s, a as c, g as p } from "./create-recipe-D2XybSAv.mjs";
import { c as b } from "./create-style-context-8AvEXXie.mjs";
const o = {}, g = [], I = [
  ["root", "number-input__root"],
  ["field", "number-input__field"],
  ["incrementTrigger", "number-input__incrementTrigger"],
  ["decrementTrigger", "number-input__decrementTrigger"]
], _ = /* @__PURE__ */ I.map(([e, n]) => [
  e,
  c(
    n,
    o,
    p(g, e)
  )
]), l = i((e = {}) => Object.fromEntries(
  _.map(([n, u]) => [n, u.recipeFn(e)])
)), m = [], d = (e) => ({ ...o, ...a(e) }), T = /* @__PURE__ */ Object.assign(l, {
  __recipe__: !1,
  __name__: "numberInput",
  raw: (e) => e,
  classNameMap: {},
  variantKeys: m,
  variantMap: {},
  splitVariantProps(e) {
    return s(e, m);
  },
  getVariantProps: d
}), { withRootProvider: f, withContext: t } = b(T), N = f(r.Root), S = t(r.Input, "field"), V = t(
  r.IncrementTrigger,
  "incrementTrigger"
), F = t(
  r.DecrementTrigger,
  "decrementTrigger"
), R = t(r.Scrubber, "field"), v = {
  Root: N,
  Field: S,
  IncrementTrigger: V,
  DecrementTrigger: F,
  Scrubber: R
};
export {
  v as N,
  N as a,
  S as b,
  V as c,
  F as d,
  R as e
};
