// liquidify-react/dist/libs/components/chunks/pagination-DDxJrXkb.mjs
import { Pagination as t } from "@ark-ui/react";
import { m as s, c as g, s as p, a as c, g as m } from "./create-recipe-D2XybSAv.mjs";
import { c as l } from "./create-style-context-8AvEXXie.mjs";
const e = {}, _ = [], P = [
  [
    "root",
    "pagination__root"
  ],
  [
    "item",
    "pagination__item"
  ],
  [
    "ellipsis",
    "pagination__ellipsis"
  ],
  [
    "prevTrigger",
    "pagination__prevTrigger"
  ],
  [
    "nextTrigger",
    "pagination__nextTrigger"
  ]
], T = /* @__PURE__ */ P.map(([i, n]) => [i, c(n, e, m(_, i))]), v = s((i = {}) => Object.fromEntries(T.map(([n, r]) => [n, r.recipeFn(i)]))), o = [], x = (i) => ({ ...e, ...g(i) }), f = /* @__PURE__ */ Object.assign(v, {
  __recipe__: !1,
  __name__: "pagination",
  raw: (i) => i,
  classNameMap: {},
  variantKeys: o,
  variantMap: {},
  splitVariantProps(i) {
    return p(i, o);
  },
  getVariantProps: x
}), { withRootProvider: V, withContext: a } = l(f), u = V(t.Root), R = a(t.Item, "item"), d = a(t.Ellipsis, "ellipsis"), C = a(t.PrevTrigger, "prevTrigger"), E = a(t.NextTrigger, "nextTrigger"), w = {
  Root: u,
  Item: R,
  Ellipsis: d,
  PrevTrigger: C,
  NextTrigger: E
};
export {
  w as P,
  u as a,
  R as b,
  d as c,
  C as d,
  E as e
};
