// liquidify-react/dist/libs/components/chunks/pinInput-Bp5Xq78T.mjs
import { jsxs as V, jsx as s } from "react/jsx-runtime";
import { PinInput as t } from "@ark-ui/react";
import { forwardRef as g } from "react";
import { m as y, a as C, s as j, b as R } from "./create-recipe-D2XybSAv.mjs";
const p = /* @__PURE__ */ C(
  "pin-input",
  {
    size: "md"
  },
  []
), m = {
  size: ["sm", "md", "lg"]
}, e = Object.keys(m), o = /* @__PURE__ */ Object.assign(y(p.recipeFn), {
  __recipe__: !0,
  __name__: "pinInput",
  __getCompoundVariantCss__: p.__getCompoundVariantCss__,
  raw: (n) => n,
  variantKeys: e,
  variantMap: m,
  merge(n) {
    return R(this, n);
  },
  splitVariantProps(n) {
    return j(n, e);
  },
  getVariantProps: p.getVariantProps
}), b = g(
  ({ label: n, length: i = 4, size: u, children: c, className: _, ...P }, I) => {
    const [d, f] = o.splitVariantProps({ size: u }), l = Array.from({ length: i }, (a, r) => `input-${r}-of-${i}`);
    return /* @__PURE__ */ V(t.Root, { ref: I, className: _, ...f, ...P, children: [
      n && /* @__PURE__ */ s(t.Label, { children: n }),
      /* @__PURE__ */ s(t.Control, { children: l.map((a, r) => /* @__PURE__ */ s(t.Input, { index: r, className: o(d) }, a)) }),
      c
    ] });
  }
);
b.displayName = "PinInput";
export {
  b as P
};
