// liquidify-react/dist/libs/components/chunks/popover-xJ1Ulzkk.mjs
import { Popover as e } from "@ark-ui/react";
import { m as n, c as a, s as c, a as g, g as v } from "./create-recipe-D2XybSAv.mjs";
import { c as l } from "./create-style-context-8AvEXXie.mjs";
const p = {}, _ = [], P = [
  [
    "trigger",
    "popover__trigger"
  ],
  [
    "positioner",
    "popover__positioner"
  ],
  [
    "content",
    "popover__content"
  ],
  [
    "title",
    "popover__title"
  ],
  [
    "description",
    "popover__description"
  ],
  [
    "closeTrigger",
    "popover__closeTrigger"
  ]
], m = /* @__PURE__ */ P.map(([o, t]) => [o, g(t, p, v(_, o))]), T = n((o = {}) => Object.fromEntries(m.map(([t, i]) => [t, i.recipeFn(o)]))), s = [], C = (o) => ({ ...p, ...a(o) }), d = /* @__PURE__ */ Object.assign(T, {
  __recipe__: !1,
  __name__: "popover",
  raw: (o) => o,
  classNameMap: {},
  variantKeys: s,
  variantMap: {},
  splitVariantProps(o) {
    return c(o, s);
  },
  getVariantProps: C
}), { withRootProvider: f, withContext: r } = l(d), V = f(e.Root), u = r(e.Trigger, "trigger"), R = r(e.Positioner, "positioner"), D = r(e.Content, "content"), S = r(e.Title, "title"), b = r(e.Description, "description"), w = r(e.CloseTrigger, "closeTrigger"), h = {
  Root: V,
  Trigger: u,
  Positioner: R,
  Content: D,
  Title: S,
  Description: b,
  CloseTrigger: w
};
export {
  h as P,
  V as a,
  u as b,
  R as c,
  D as d,
  S as e,
  b as f,
  w as g
};
