// liquidify-react/dist/libs/components/chunks/progress-C0-XJBvJ.mjs
import { jsxs as d, jsx as e } from "react/jsx-runtime";
import { Progress as s } from "@ark-ui/react";
import { forwardRef as f } from "react";
import { m as u, a as j, s as R, b as x } from "./create-recipe-D2XybSAv.mjs";
const a = /* @__PURE__ */ j("progress", {
  size: "md"
}, []), n = {
  size: [
    "sm",
    "md",
    "lg"
  ]
}, t = Object.keys(n), i = /* @__PURE__ */ Object.assign(u(a.recipeFn), {
  __recipe__: !0,
  __name__: "progress",
  __getCompoundVariantCss__: a.__getCompoundVariantCss__,
  raw: (r) => r,
  variantKeys: t,
  variantMap: n,
  merge(r) {
    return x(this, r);
  },
  splitVariantProps(r) {
    return R(r, t);
  },
  getVariantProps: a.getVariantProps
}), y = f(
  ({ label: r, value: p = 0, max: m = 100, size: g, children: o, className: c, ..._ }, P) => {
    const [l, V] = i.splitVariantProps({ size: g });
    return /* @__PURE__ */ d(
      s.Root,
      {
        ref: P,
        className: [i(l), c].filter(Boolean).join(" "),
        value: p,
        max: m,
        ...V,
        ..._,
        children: [
          (r || o) && /* @__PURE__ */ e(s.Label, { children: r || o }),
          /* @__PURE__ */ e(s.Track, { children: /* @__PURE__ */ e(s.Range, {}) }),
          /* @__PURE__ */ e(s.ValueText, {})
        ]
      }
    );
  }
);
y.displayName = "Progress";
export {
  y as P
};
