// liquidify-react/dist/libs/components/chunks/radioGroup-DPqyWjaM.mjs
import { RadioGroup as t } from "@ark-ui/react";
import { m as n, c as p, s as m, a as c, g as u } from "./create-recipe-D2XybSAv.mjs";
import { c as d } from "./create-style-context-8AvEXXie.mjs";
const i = {}, _ = [], G = [
  ["root", "radio-group__root"],
  ["item", "radio-group__item"],
  ["itemControl", "radio-group__itemControl"],
  ["itemText", "radio-group__itemText"]
], l = /* @__PURE__ */ G.map(([o, r]) => [
  o,
  c(
    r,
    i,
    u(_, o)
  )
]), R = n((o = {}) => Object.fromEntries(
  l.map(([r, s]) => [r, s.recipeFn(o)])
)), e = [], C = (o) => ({ ...i, ...p(o) }), x = /* @__PURE__ */ Object.assign(R, {
  __recipe__: !1,
  __name__: "radioGroup",
  raw: (o) => o,
  classNameMap: {},
  variantKeys: e,
  variantMap: {},
  splitVariantProps(o) {
    return m(o, e);
  },
  getVariantProps: C
}), { withRootProvider: I, withContext: a } = d(x), g = I(t.Root), f = a(t.Item, "item"), T = a(t.ItemControl, "itemControl"), V = a(t.ItemText, "itemText"), v = {
  Root: g,
  Item: f,
  ItemControl: T,
  ItemText: V
};
export {
  v as R,
  g as a,
  f as b,
  T as c,
  V as d
};
