// liquidify-react/dist/libs/components/chunks/ratingGroup-Cns5ZGZD.mjs
import { jsxs as h, jsx as t } from "react/jsx-runtime";
import { RatingGroup as a } from "@ark-ui/react";
import { forwardRef as V } from "react";
import { m as f, a as C, s as P, b as x } from "./create-recipe-D2XybSAv.mjs";
const n = /* @__PURE__ */ C(
  "rating-group",
  {
    size: "md"
  },
  []
), s = {
  size: ["sm", "md", "lg"]
}, o = Object.keys(s), i = /* @__PURE__ */ Object.assign(f(n.recipeFn), {
  __recipe__: !0,
  __name__: "ratingGroup",
  __getCompoundVariantCss__: n.__getCompoundVariantCss__,
  raw: (r) => r,
  variantKeys: o,
  variantMap: s,
  merge(r) {
    return x(this, r);
  },
  splitVariantProps(r) {
    return P(r, o);
  },
  getVariantProps: n.getVariantProps
}), j = V(
  ({ label: r, count: p = 5, size: m, children: c, className: g, ...u }, _) => {
    const [l, d] = i.splitVariantProps({ size: m });
    return /* @__PURE__ */ h(
      a.Root,
      {
        ref: _,
        className: [i(l), g].filter(Boolean).join(" "),
        count: p,
        ...d,
        ...u,
        children: [
          r && /* @__PURE__ */ t(a.Label, { children: r }),
          /* @__PURE__ */ t(a.Control, { children: /* @__PURE__ */ t(a.Context, { children: ({ items: G }) => G.map((e) => /* @__PURE__ */ t(a.Item, { index: e, children: /* @__PURE__ */ t(a.ItemContext, { children: ({ highlighted: R }) => /* @__PURE__ */ t("span", { children: R ? "★" : "☆" }) }) }, e)) }) }),
          c
        ]
      }
    );
  }
);
j.displayName = "RatingGroup";
export {
  j as R
};
