// liquidify-react/dist/libs/components/chunks/select-CKaHWigB.mjs
import { Select as t } from "@ark-ui/react";
import { m as a, c as i, s as l, a as g, g as m } from "./create-recipe-D2XybSAv.mjs";
import { c as p } from "./create-style-context-8AvEXXie.mjs";
const c = {}, _ = [], S = [
  [
    "trigger",
    "select__trigger"
  ],
  [
    "content",
    "select__content"
  ],
  [
    "item",
    "select__item"
  ],
  [
    "itemText",
    "select__itemText"
  ],
  [
    "positioner",
    "select__positioner"
  ],
  [
    "indicator",
    "select__indicator"
  ],
  [
    "clearTrigger",
    "select__clearTrigger"
  ]
], T = /* @__PURE__ */ S.map(([e, o]) => [e, g(o, c, m(_, e))]), d = a((e = {}) => Object.fromEntries(T.map(([o, n]) => [o, n.recipeFn(e)]))), s = [], C = (e) => ({ ...c, ...i(e) }), x = /* @__PURE__ */ Object.assign(d, {
  __recipe__: !1,
  __name__: "select",
  raw: (e) => e,
  classNameMap: {},
  variantKeys: s,
  variantMap: {},
  splitVariantProps(e) {
    return l(e, s);
  },
  getVariantProps: C
}), { withRootProvider: I, withContext: r } = p(x), f = I(t.Root), P = r(t.Trigger, "trigger"), V = r(t.Positioner, "positioner"), u = r(t.Content, "content"), R = r(t.Item, "item"), b = r(t.ItemText, "itemText"), h = r(t.Indicator, "indicator"), v = r(t.ClearTrigger, "clearTrigger"), j = {
  Root: f,
  Trigger: P,
  Positioner: V,
  Content: u,
  Item: R,
  ItemText: b,
  Indicator: h,
  ClearTrigger: v
};
export {
  j as S,
  f as a,
  P as b,
  V as c,
  u as d,
  R as e,
  b as f,
  h as g,
  v as h
};
