// liquidify-react/dist/libs/components/chunks/slider-5l750WR3.mjs
import { jsxs as t, jsx as s } from "react/jsx-runtime";
import { Slider as e } from "@ark-ui/react";
import { forwardRef as V } from "react";
import { m as f, a as u, s as P, b as h } from "./create-recipe-D2XybSAv.mjs";
const a = /* @__PURE__ */ u("slider", {
  size: "md"
}, []), p = {
  size: [
    "sm",
    "md",
    "lg"
  ]
}, o = Object.keys(p), n = /* @__PURE__ */ Object.assign(f(a.recipeFn), {
  __recipe__: !0,
  __name__: "slider",
  __getCompoundVariantCss__: a.__getCompoundVariantCss__,
  raw: (r) => r,
  variantKeys: o,
  variantMap: p,
  merge(r) {
    return h(this, r);
  },
  splitVariantProps(r) {
    return P(r, o);
  },
  getVariantProps: a.getVariantProps
}), j = V(
  ({ label: r, size: m, children: i, className: l, ...d }, c) => {
    const [_, g] = n.splitVariantProps({ size: m });
    return /* @__PURE__ */ t(
      e.Root,
      {
        ref: c,
        className: [n(_), l].filter(Boolean).join(" "),
        ...g,
        ...d,
        children: [
          (r || i) && /* @__PURE__ */ s(e.Label, { children: r || i }),
          /* @__PURE__ */ t(e.Control, { children: [
            /* @__PURE__ */ s(e.Track, { children: /* @__PURE__ */ s(e.Range, {}) }),
            /* @__PURE__ */ s(e.Thumb, { index: 0 })
          ] })
        ]
      }
    );
  }
);
j.displayName = "Slider";
export {
  j as S
};
