// liquidify-react/dist/libs/components/chunks/tabs-BRG6cg5O.mjs
import { Tabs as a } from "@ark-ui/react";
import { m as i, c, s as b, a as p, g as _ } from "./create-recipe-D2XybSAv.mjs";
import { c as m } from "./create-style-context-8AvEXXie.mjs";
const r = {}, g = [], T = [
  [
    "root",
    "tabs__root"
  ],
  [
    "list",
    "tabs__list"
  ],
  [
    "trigger",
    "tabs__trigger"
  ],
  [
    "content",
    "tabs__content"
  ],
  [
    "indicator",
    "tabs__indicator"
  ]
], l = /* @__PURE__ */ T.map(([t, o]) => [t, p(o, r, _(g, t))]), d = i((t = {}) => Object.fromEntries(l.map(([o, e]) => [o, e.recipeFn(t)]))), n = [], C = (t) => ({ ...r, ...c(t) }), f = /* @__PURE__ */ Object.assign(d, {
  __recipe__: !1,
  __name__: "tabs",
  raw: (t) => t,
  classNameMap: {},
  variantKeys: n,
  variantMap: {},
  splitVariantProps(t) {
    return b(t, n);
  },
  getVariantProps: C
}), { withRootProvider: V, withContext: s } = m(f), u = V(a.Root), R = s(a.List, "list"), P = s(a.Trigger, "trigger"), S = s(a.Content, "content"), v = s(a.Indicator, "indicator"), F = {
  Root: u,
  List: R,
  Trigger: P,
  Content: S,
  Indicator: v
};
export {
  F as T,
  u as a,
  R as b,
  P as c,
  S as d,
  v as e
};
