// liquidify-react/dist/libs/components/chunks/toast-DRWpNNoo.mjs
import { Toast as o } from "@ark-ui/react";
import { m as n, c, s as g, a as p, g as T } from "./create-recipe-D2XybSAv.mjs";
import { c as l } from "./create-style-context-8AvEXXie.mjs";
const e = {
  status: "info"
}, _ = [], m = [
  [
    "root",
    "toast__root"
  ],
  [
    "title",
    "toast__title"
  ],
  [
    "description",
    "toast__description"
  ],
  [
    "closeTrigger",
    "toast__closeTrigger"
  ],
  [
    "actionTrigger",
    "toast__actionTrigger"
  ]
], u = /* @__PURE__ */ m.map(([t, a]) => [t, p(a, e, T(_, t))]), f = n((t = {}) => Object.fromEntries(u.map(([a, i]) => [a, i.recipeFn(t)]))), r = [
  "status"
], C = (t) => ({ ...e, ...c(t) }), d = /* @__PURE__ */ Object.assign(f, {
  __recipe__: !1,
  __name__: "toast",
  raw: (t) => t,
  classNameMap: {},
  variantKeys: r,
  variantMap: {
    status: [
      "info",
      "success",
      "warning",
      "error"
    ]
  },
  splitVariantProps(t) {
    return g(t, r);
  },
  getVariantProps: C
}), { withContext: s } = l(d), V = s(o.Root, "root"), D = s(o.Title, "title"), R = s(o.Description, "description"), S = s(o.CloseTrigger, "closeTrigger"), b = s(o.ActionTrigger, "actionTrigger"), A = {
  Root: V,
  Title: D,
  Description: R,
  CloseTrigger: S,
  ActionTrigger: b
};
export {
  A as T,
  V as a,
  D as b,
  R as c,
  S as d,
  b as e
};
