// liquidify-react/dist/libs/components/chunks/tooltip-CiJ3uvhB.mjs
import { Tooltip as o } from "@ark-ui/react";
import { m as a, c as p, s as c, a as l, g as m } from "./create-recipe-D2XybSAv.mjs";
import { c as g } from "./create-style-context-8AvEXXie.mjs";
const s = {}, _ = [], T = [
  [
    "trigger",
    "tooltip__trigger"
  ],
  [
    "positioner",
    "tooltip__positioner"
  ],
  [
    "content",
    "tooltip__content"
  ]
], C = /* @__PURE__ */ T.map(([t, i]) => [t, l(i, s, m(_, t))]), P = a((t = {}) => Object.fromEntries(C.map(([i, e]) => [i, e.recipeFn(t)]))), r = [], f = (t) => ({ ...s, ...p(t) }), V = /* @__PURE__ */ Object.assign(P, {
  __recipe__: !1,
  __name__: "tooltip",
  raw: (t) => t,
  classNameMap: {},
  variantKeys: r,
  variantMap: {},
  splitVariantProps(t) {
    return c(t, r);
  },
  getVariantProps: f
}), { withRootProvider: u, withContext: n } = g(V), R = u(o.Root), d = n(o.Content, "content"), S = n(o.Positioner, "positioner"), b = n(o.Trigger, "trigger"), y = {
  Root: R,
  Content: d,
  Positioner: S,
  Trigger: b
};
export {
  y as T,
  R as a,
  d as b,
  S as c,
  b as d
};
